app/controllers/carto/api/visualizations_controller.rb

```diff
@@ -355,12 +355,16 @@ def count
         type = params.fetch(:type, 'datasets')
         typeList = (type == 'datasets') ? "'table','remote'" : "'derived'"
         categoryType = (type == 'datasets') ? 1 : 2
+        samples_user_id = sample_maps_user.id if categoryType == 2 && sample_maps_user
+        sample_type_counts = Object.new
```

Comment:
Initializing a variable like this isn't to common in Ruby. If you must, you can initialize to `nil`, and then invoke it as

```ruby
sample_type_counts || default_value
```

and set it like
```ruby
sample_type_counts ||= "whatever overrite if not set"
```